modules/juce_audio_formats/codecs/juce_CoreAudioFormat.cpp

```diff
@@ -383,229 +387,155 @@ struct CoreAudioFormatMetatdata
+                      juce::int64 startSampleInFile, int numSamples) override
     {
         clearSamplesBeyondAvailableLength (destSamples, numDestChannels, startOffsetInDestBuffer,
                                            startSampleInFile, numSamples, lengthInSamples);
 
-        if (numSamples <= 0)
-            return true;
-
-        if (lastReadPosition != startSampleInFile)
+        if (audioFile == nil)
         {
-            OSStatus status = ExtAudioFileSeek (audioFileRef, startSampleInFile);
-            if (status != noErr)
-                return false;
-
-            lastReadPosition = startSampleInFile;
+            NSLog(@"Audio file is nil");
+            return false;
         }
 
-        while (numSamples > 0)
+        if (!reusableBuffer ||
+            ![reusableBuffer.format isEqual:audioFile.processingFormat] ||
+            reusableBuffer.frameCapacity < numSamples)
         {
-            auto numThisTime = jmin (8192, numSamples);
-            auto numBytes = (size_t) numThisTime * sizeof (float);
-
-            audioDataBlock.ensureSize (numBytes * numChannels, false);
-            auto* data = static_cast<float*> (audioDataBlock.getData());
-
-            for (int j = (int) numChannels; --j >= 0;)
-            {
-                bufferList->mBuffers[j].mNumberChannels = 1;
-                bufferList->mBuffers[j].mDataByteSize = (UInt32) numBytes;
-                bufferList->mBuffers[j].mData = data;
-                data += numThisTime;
-            }
+            NSLog(@"Creating new reusable buffer");
+            reusableBuffer = [[AVAudioPCMBuffer alloc] initWithPCMFormat:audioFile.processingFormat frameCapacity:numSamples];
+        }
 
-            auto numFramesToRead = (UInt32) numThisTime;
-            auto status = ExtAudioFileRead (audioFileRef, &numFramesToRead, bufferList);
+        // Now check if startSampleInFile is the same as the current file position, if not change it
+        if (audioFile.framePosition != startSampleInFile)
+        {
+            audioFile.framePosition = startSampleInFile;
+        }
 
-            if (status != noErr)
-                return false;
+        NSError* error = nil;
+        [audioFile readIntoBuffer:reusableBuffer frameCount:numSamples error:&error];
+        if (error)
+        {
+            NSLog(@"Error reading audio file: %@", error);
+            return false;
+        }
 
-            if (numFramesToRead == 0)
-                break;
+        const auto numBytes = static_cast<size_t>(numSamples * sizeof(float));
 
-            if ((int) numFramesToRead < numThisTime)
-            {
-                numThisTime = (int) numFramesToRead;
-                numBytes    = (size_t) numThisTime * sizeof (float);
-            }
+        // Also mostly copied from previous JUCE Reader
+        for (int i = numDestChannels; --i >= 0;)
+        {
+            auto* dest = destSamples[(i < static_cast<int>(numChannels) ? channelMap[i] : i)];

```

Comment:
const?

Reply:
dest can't be as it's getting stuff copied into it